test/hotspot/jtreg/containers/docker/TestJFREvents.java

```diff
@@ -70,31 +70,116 @@ public static void main(String[] args) throws Exception {
-        Common.logNewTestCase("CPUInfo: --cpus = " + valueToSet);
-        String fieldName = "activeProcessorCount";
+    private static void testContainerInfo(int expectedCPUs, int expectedMemoryMB) throws Exception {
+        Common.logNewTestCase("ContainerInfo: --cpus = " + expectedCPUs + " --memory=" + expectedMemoryMB + "m");
+        String eventName = "jdk.ContainerConfiguration";
+        long expectedSlicePeriod = 100000; // default slice period
+        long expectedMemoryLimit = expectedMemoryMB * 1024 * 1024;
+
+        String cpuCountFld = "effectiveCpuCount";
+        String cpuQuotaFld = "cpuQuota";
+        String cpuSlicePeriodFld = "cpuSlicePeriod";
+        String memoryLimitFld = "memoryLimit";
+
         DockerTestUtils.dockerRunJava(
                                       commonDockerOpts()
-                                      .addDockerOpts("--cpus=" + valueToSet)
+                                      .addDockerOpts("--cpus=" + expectedCPUs)
+                                      .addDockerOpts("--memory=" + expectedMemoryMB + "m")
                                       .addClassOptions(eventName))
             .shouldHaveExitValue(0)
-            .shouldContain(fieldName + " = " + expectedValue);
+            .shouldContain(cpuCountFld + " = " + expectedCPUs)
+            .shouldContain(cpuSlicePeriodFld + " = " + expectedSlicePeriod)
+            .shouldContain(cpuQuotaFld + " = " + expectedCPUs * expectedSlicePeriod)
+            .shouldContain(memoryLimitFld + " = " + expectedMemoryLimit);
+    }
+
+    private static void testCpuUsage() throws Exception {
+        Common.logNewTestCase("CPU Usage");
+        String eventName = "jdk.ContainerCPUUsage";
+
+        String cpuTimeFld = "cpuTime";
+        String cpuUserTimeFld = "cpuUserTime";
+        String cpuSystemTimeFld = "cpuSystemTime";
+
+        DockerTestUtils.dockerRunJava(
+                                      commonDockerOpts()
+                                      .addClassOptions(eventName, "period=endChunk"))
+            .shouldHaveExitValue(0)
+            .shouldNotContain(cpuTimeFld + " = " + 0)
+            .shouldNotContain(cpuUserTimeFld + " = " + 0)
+            .shouldNotContain(cpuSystemTimeFld + " = " + 0);
+    }
+
+    private static void testMemoryUsage() throws Exception {
+        Common.logNewTestCase("Memory Usage");
+        String eventName = "jdk.ContainerMemoryUsage";
+
+        String memoryPressureFld = "memoryPressure";
+        String memoryFailCountFld = "memoryFailCount";
+        String memoryUsageFld = "memoryUsage";
+        String swapMemoryUsageFld = "swapMemoryUsage";
+
+        DockerTestUtils.dockerRunJava(
+                                      commonDockerOpts()
+                                      .addClassOptions(eventName, "period=endChunk"))
+            .shouldHaveExitValue(0)
+            .shouldContain(memoryPressureFld)
```

Comment:
This test fails for me on cgroupv1 with:

```
----------System.err:(42/1407)----------
 stdout: [===== EventType: jdk.ContainerMemoryUsage
startTime = 946400166
duration = 0
eventThread = {
  osName = "main"
  osThreadId = 6
  javaName = "main"
  javaThreadId = 1
  group = {
    parent = {
      parent = N/A
      name = "system"
    }
    name = "main"
  }
}

stackTrace = null
memoryFailCount = 0
memoryUsage = 57786368
swapMemoryUsage = 57782272
];
 stderr: []
 exitValue = 0

java.lang.RuntimeException: 'memoryPressure' missing from stdout/stderr

        at jdk.test.lib.process.OutputAnalyzer.shouldContain(OutputAnalyzer.java:206)
        at TestJFREvents.testMemoryUsage(TestJFREvents.java:147)
        at TestJFREvents.main(TestJFREvents.java:77)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at com.sun.javatest.regtest.agent.MainWrapper$MainThread.run(MainWrapper.java:127)
        at java.base/java.lang.Thread.run(Thread.java:831)

JavaTest Message: Test threw exception: java.lang.RuntimeException: 'memoryPressure' missing from stdout/stderr

JavaTest Message: shutting down test
```

I think `memoryPressure` got removed from the code and, thus, should get removed from the test.

Reply:
Will fix